Source/API/Extras/CBLIncrementalStore.m

```diff
@@ -761,239 +634,797 @@ - (void) _initializeViews
+                    }
+                } else {
+                    NSString *parentDocId = [properties objectForKey:@"_id"];
+                    if (parentDocId) {
+                        CBLQueryEnumerator *rows = [self queryToManyRelation:relation
+                                                               forParentKeys:@[parentDocId]
+                                                                    prefetch:YES
+                                                                    outError:nil];
+                        if (rows) {
+                            NSMutableArray *values = [NSMutableArray array];
+                            for (CBLQueryRow *row in rows) {
+                                id propValue = row.documentProperties[destKeyPath];
+                                if (propValue)
+                                    [values addObject:propValue];
+                            }
+                            value = values;
+                        }
+                    }
+                }
+            }
         }
+            break;
+        default:
+            NSAssert(NO, @"[devel] Expression Type not yet supported: %@", expression);
+            break;
     }
-    return result;
+
+    // not supported yet:
+    //    NSFunctionExpressionType,
+    //    NSAggregateExpressionType,
+    //    NSSubqueryExpressionType = 13,
+    //    NSUnionSetExpressionType,
+    //    NSIntersectSetExpressionType,
+    //    NSMinusSetExpressionType,
+    //    NSBlockExpressionType = 19
+    
+    return value;
 }
 
-- (NSManagedObjectID *)_newObjectIDForEntity:(NSEntityDescription *)entity managedObjectContext:(NSManagedObjectContext*)context
-                                     couchID:(NSString*)couchID
-{
-    NSManagedObjectID *objectID = [self newObjectIDForEntity:entity referenceObject:couchID];
-    return objectID;
+- (CBLQueryEnumerator *)queryToManyRelation:(NSRelationshipDescription *)relation
+                              forParentKeys:(NSArray *)parentKeys
+                                   prefetch:(BOOL)prefetch
+                                   outError:(NSError **)outError {
+    CBLView *view = [self.database existingViewNamed:CBLISToManyViewNameForRelationship(relation)];
+    if (view) {
+        CBLQuery* query = [view createQuery];
+        query.keys = parentKeys;
+        query.prefetch = prefetch;
+        return [self _queryEnumeratorForQuery:query error:outError];
+    }
+    return nil;
 }
 
-- (id) _couchbaseLiteRepresentationOfManagedObject:(NSManagedObject*)object withCouchbaseLiteID:(BOOL)withID
-{
+#pragma mark - Misc
+
+- (id) _couchbaseLiteValueForAttributeValue:(id)value ofType:(NSAttributeType)type {
+    id result = nil;
+
+    switch (type) {
+        case NSInteger16AttributeType:
+        case NSInteger32AttributeType:
+        case NSInteger64AttributeType:
+            result = [NSNumber numberWithLong:CBLISIsNull(value) ? 0 : [value longValue]];
```

Comment:
I agree that we don't need to create a new NSNumber of it's already NSNumber. This is from the original code so I will run some tests to make sure that it will not break anything.

Regarding the data type check, as the value is coming from CoreData, I think we could expect that the value has a valid type. But when converting from CBLDocument to NSManagedObject , I think it should validate the type there.